spec/services/push_notifications/send_spec.rb

```diff
@@ -0,0 +1,59 @@
+require "rails_helper"
+
+RSpec.describe PushNotifications::Send, type: :service do
+  let(:user) { create(:user) }
+  let(:params) do
+    {
+      user: user,
+      title: "Alert",
+      body: "some alert here",
+      payload: ""
+    }
+  end
+
+  context "with feature flag set to false/off" do
+    before { allow(FeatureFlag).to receive(:enabled?).with(:mobile_notifications).and_return(false) }
+
+    it "does nothing if the feature flag is disabled" do
+      expect do
+        described_class.call(params)
+      end.not_to change { Rpush::Client::Redis::Notification.all.count }
+    end
+  end
+
+  context "with no devices for user" do
+    before { allow(FeatureFlag).to receive(:enabled?).with(:mobile_notifications).and_return(true) }
+
+    it "does nothing" do
+      expect(user.devices.count).to eq(0)
+      expect do
+        described_class.call(params)
+      end.not_to change { Rpush::Client::Redis::Notification.all.count }
+    end
+  end
+
+  context "with devices for user" do
+    before do
+      allow(FeatureFlag).to receive(:enabled?).with(:mobile_notifications).and_return(true)
+      allow(ApplicationConfig).to receive(:[]).with("RPUSH_IOS_PEM").and_return("dGVzdGluZw==")
+      allow(ApplicationConfig).to receive(:[]).with("COMMUNITY_NAME").and_return("Forem")
+      create(:device, user: user)
+    end
+
+    it "creates a notification and enqueues it" do
+      expect do
+        described_class.call(params)
+      end.to change { Rpush::Client::Redis::Notification.all.count }.by(1)
```

Comment:
Nitpicky, but I'd find this easier to read if we move this call down too:

```ruby
expect { described_class.call(params) }
  .to change { Rpush::Client::Redis::Notification.all.count }.by(1)
  .and change(PushNotifications::DeliverWorker.jobs, :size).by(1)
```

Reply:
✅